File: library/src/main/java/su/levenetc/android/textsurface/common/ScaleValue.java
package su.levenetc.android.textsurface.common;

import android.graphics.PointF;

/**
 * Created by Eugene Levenetc.
 */
public class ScaleValue {

	private PointF scale = new PointF(1, 1);
	private PointF pivot = new PointF();

	public void setValue(float scale) {
		this.scale.set(scale, scale);
	}

	public void setValueX(float x) {
		scale.set(x, scale.y);
	}

	public void setValueY(float y) {
		scale.set(scale.x, y);
	}

	public float getScaleX() {
		return scale.x;
	}

	public float getScaleY() {
		return scale.y;
	}

	public PointF getPivot() {
		return pivot;
	}

	public void reset() {
		scale.set(1, 1);
		pivot.set(0, 0);
	}
}

File: library/src/main/java/su/levenetc/android/textsurface/common/ScaleValue.kt
package su.levenetc.android.textsurface.common

import android.graphics.PointF

/**
 * Created by Eugene Levenetc.
 */
data class ScaleValue(val scale: PointF = PointF(1f, 1f), val pivot: PointF = PointF()) {

    fun setValue(scale: Float) {
        this.scale[scale] = scale
    }

    fun setValueX(x: Float) {
        scale[x] = scale.y
    }

    fun setValueY(y: Float) {
        scale[scale.x] = y
    }

    val scaleX: Float
        get() = scale.x
    val scaleY: Float
        get() = scale.y

    fun reset() {
        scale[1f] = 1f
        pivot[0f] = 0f
    }
}